File: ios/FHIRmHealth/HealthKitQuery.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(HealthKitQuery, NSObject)

RCT_EXTERN_METHOD(activitySummary:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

+(BOOL)requiresMainQueueSetup {
  return NO;
}

@end

File: ios/FHIRmHealth/HealthKitQuery.swift
@objc(HealthKitQuery)
class HealthKitQuery: NSObject {
  @objc func activitySummary(_ resolve: @escaping RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
    do {
      try HealthKitConnector.shared.queryActivitySummary { activitySummary in
        resolve(activitySummary?.serialized)
      }
    } catch {
      reject("ERR_QUERY_FAILED", "Can not query for latest activity summary", error)
    }
  }
}
